### ISSUE TITLE
Mappers should all take arrays as inputs, as well as fields

### ISSUE THREAD

**Issue by SiccarPoint** [landlab/landlab#260]
It should be possible to do:
```
>>> z = mg.ones('node')
>>> link_z = mg.map_mean_of_link_nodes_to_link(z)
```
...as well as the only way we have now:
```
>>> mg.add_ones('node', 'topo')
>>> link_z = mg.map_mean_of_link_nodes_to_link('topo')
```

Plenty of cases where you need to map "on the fly", and it doesn't warrant creation of a field.

**Comment by SiccarPoint** [comment]
I think there's some consistency/documentation issues too in terms of whether output fields are created. Various of the mappers e.g.,mg.map_max_of_outlinks_to_node() make claims to "...[return] a field at the nodes with the same var_name as the link field", whereas I don't think any new field is created - they actually just return an array.

**Comment by gregtucker** [comment]
Yes! to resolving this issue. It would be very helpful.

**Comment by SiccarPoint** [comment]
I'm on this right now

**Comment by SiccarPoint** [comment]
Resolved by #263
